chore(fe): Convert `<ReleasesList />` to an FC

leeandher opened the pull request:
It's a pretty big rewrite, but here are the callouts:
- Importing the deprecated ~~`withSentryRouter`~~ `useRouter()` for the inner list component. This is just so we don't have to radically change two components at once, and can focus on the AsyncComponent in this PR
- The `getSelectedProjectSlugs` and `projectHasSession` methods were unused.
- In the test file, the one 'empty state' test has been broken up into 4 to avoid excessive unmount/rerender loops, especially since the new syntax for location doesn't work with `rerender`


For testing this one out, I kinda just clicked around in `dev-ui` and double checked that all the existing behavior was maintained. If you notice anything wrong on the deploy preview let me know :+1:

leeandher committed:
♻️ Convert releases list to FC

leeandher committed:
Test rewrites for fc

leeandher committed:
♻️ Fix some TS types and bring forcecheck

ryan953 commented on static/app/views/releases/list/releaseListInner.tsx:
```diff
@@ -130,6 +134,8 @@ export default function ReleaseListInner({
   );
 }
 
+export default withSentryRouter(ReleaseListInner);
```
could be better to call `const router = useRouter();` inside the component, and drop the `router: InjectedRouter` at the same time instead of adding an HoC. simpler to replace the useRouter call later.

leeandher committed:
♻️ Forgot about hooks

billyvg commented:
<img width="260" alt="image" src="https://github.com/user-attachments/assets/8d7d1f67-3a83-4d2b-b5a0-71d5d241c017" />
changing this from 24h to 14d doesn't seem to re-render page? on prod I see crash free rate changing

leeandher committed:
♻️ fix reloading for health stats

leeandher commented:
> changing this from 24h to 14d doesn't seem to re-render page? on prod I see crash free rate changing
Fixed I believe. The sessions stat request was reloading because it was remounted every time, but now were just passing new query params to it. I removed the `location` exception and it works fine now.

The loading/reloading state is different though. It looks like we hold on to the previous value of `releases` and make it appear disabled while we refetch, which is kinda nicer than just showing a full page spinner. I'll work on getting that back before merging

leeandher committed:
✨ better looking refetches

leeandher commented:
> The loading/reloading state is different though. It looks like we hold on to the previous value of releases and make it appear disabled while we refetch, which is kinda nicer than just showing a full page spinner.

Fixed in https://github.com/getsentry/sentry/pull/91281/commits/50deacbcf8979cf5fd6d25fa33167cbcadbe44ac with useQuery's  `placholderData` option 👍  pretty neat

leeandher committed:
Merge branch 'master' into leander/fc-releases

billyvg commented on static/app/views/releases/list/releasesRequest.tsx:
```diff
@@ -29,13 +29,7 @@ import {getCrashFreePercent} from 'sentry/views/releases/utils';
-    'organization',
-    'children',
-    'selection.datetime.utc',
-    'location',
```
Is this intentional?

leeandher replied:
Yep, removing `location` from the ignored props allows changes to the filtering to reload the health stats like crash free rate. Before it'd remount the whole thing so it'd reload anyway
